src/nvim/memory.c

```diff
@@ -812,6 +813,11 @@ void free_all_mem(void)
     bufref_T bufref;
     set_bufref(&bufref, buf);
     nextbuf = buf->b_next;
+
+    // Since options have been freed above we need to ensure no callbacks are called,
```

Comment:
This comment may be a bit strange. A lot of stuff have been freed above, not just options.

Reply:
Other being freed can cause other issues.

Reply:
I've updated the comment.